cmd/clusterctl/client/repository/repository_github.go

```diff
@@ -102,18 +106,26 @@ func (g *gitHubRepository) ComponentsPath() string {
 
 // GetFile returns a file for a given provider version.
 func (g *gitHubRepository) GetFile(version, path string) ([]byte, error) {
-	release, err := g.getReleaseByTag(version)
-	if err != nil {
-		return nil, errors.Wrapf(err, "failed to get GitHub release %s", version)
-	}
+	var files []byte
+	var retryError error
+	_ = wait.PollImmediate(retryableOperationInterval, retryableOperationTimeout, func() (bool, error) {
+		release, err := g.getReleaseByTag(version)
+		if err != nil {
+			retryError = errors.Wrapf(err, "failed to get GitHub release %s", version)
+			return false, nil
+		}
 
-	// download files from the release
-	files, err := g.downloadFilesFromRelease(release, path)
-	if err != nil {
-		return nil, errors.Wrapf(err, "failed to download files from GitHub release %s", version)
-	}
+		// download files from the release
+		var downloadError error
+		files, downloadError = g.downloadFilesFromRelease(release, path)
+		if downloadError != nil {
+			retryError = errors.Wrapf(downloadError, "failed to download files from GitHub release %s", version)
+			return false, nil
+		}
```

Comment:
Both `g.getReleaseByTag` and `g.downloadFilesFromRelease` are already wrapped in retry blocks. I dont see a benefit in wrapping this in a retry block again.

Reply:
fair enough